libp2p/network/connection/swarm_connection.py

```diff
@@ -15,21 +17,17 @@
 """
 
 
-class SwarmConn(INetConn):
+class SwarmConn(INetConn, Service):
     muxed_conn: IMuxedConn
     swarm: "Swarm"
     streams: Set[NetStream]
-    event_closed: asyncio.Event
-
-    _tasks: List["asyncio.Future[Any]"]
+    event_closed: trio.Event
 
     def __init__(self, muxed_conn: IMuxedConn, swarm: "Swarm") -> None:
         self.muxed_conn = muxed_conn
         self.swarm = swarm
         self.streams = set()
-        self.event_closed = asyncio.Event()
-
-        self._tasks = []
+        self.event_closed = trio.Event()
 
     async def close(self) -> None:
```

Comment:
This method looks out of place for a `Service`.  I would have at least expected `close()` to trigger `self.manager.cancel()`...

Reply:
Sorry, I missed these feedback previously. Good point. Will address it in the next commits.